lib/logstash/codecs/json_lines.rb

```diff
@@ -51,31 +70,21 @@ def encode(event)
   def flush(&block)
     remainder = @buffer.flush
     if !remainder.empty?
-      parse(@converter.convert(remainder), &block)
+      parse_json(@converter.convert(remainder), &block)
     end
   end
 
   private
 
-  # from_json_parse uses the Event#from_json method to deserialize and directly produce events
-  def from_json_parse(json, &block)
-    LogStash::Event.from_json(json).each { |event| yield event }
-  rescue LogStash::Json::ParserError => e
-    @logger.warn("JSON parse error, original data now in message field", :error => e, :data => json)
-    yield LogStash::Event.new("message" => json, "tags" => ["_jsonparsefailure"])
+  def parse_json(json)
```

Comment:
This method yields events, so would be more helpful to be explicit also in the parameter list including `&block`?

Reply:
an implicit block is always present - passing the `&block` and not using the `block` var makes the IDE complain :bulb: 
we do not have a style guide but I looked around and most Ruby guides do the `yield` format without a `&block` ...

Reply:
Ok, I'm fine with it!